lib/sandbox.macos.ml

```diff
@@ -127,12 +127,11 @@ let run ~cancelled ?stdin:stdin ~log (t : t) config result_tmp =
   in
   Lwt.on_termination cancelled (fun () ->
     let aux () =
-      post_cancellation ~result_tmp t >>= fun () ->
-      if Lwt.is_sleeping proc then (
-        match !proc_id with
+      if Lwt.is_sleeping proc then
+        let _ = match !proc_id with
```

Comment:
I don't understand exactly what's happening here, but it looks wrong as you're not _binding_ the Lwt promess, or am I reading that wrong?

Reply:
I have updated the binding and changed the function a bit as well.